A command like /bin/sh ../../i586-poky-linux-libtool   --mode=install /usr/bin/install -c   gck-roots-store-standalone.la '/media/data1/builds/poky1/tmp/work/core2-poky-linux/gnome-keyring-2.26.1-r1/image/usr/lib/gnome-keyring/standalone/' fails (e.g. gnome-keyring or pulseaudio)

This is because libdir has a trailing slash which breaks the comparision.

RP 2/1/10

Index: libtool-2.4/libltdl/config/ltmain.sh
===================================================================
--- libtool-2.4.orig/libltdl/config/ltmain.sh	2010-09-22 18:45:18.000000000 +-0300
+++ libtool-2.4/libltdl/config/ltmain.sh	2010-09-22 18:45:18.000000000 +-0300
@@ -2950,14 +2950,18 @@
 
 	func_dirname "$file" "/" ""
 	dir="$func_dirname_result"
 	func_append dir "$objdir"
 
 	if test -n "$relink_command"; then
+          # Strip any trailing slash from the destination.
+          func_stripname '' '/' "$libdir"
+          destlibdir=$func_stripname_result
+	
 	  # Determine the prefix the user has applied to our future dir.
-	  inst_prefix_dir=`$ECHO "$destdir" | $SED -e "s%$libdir\$%%"`
+	  inst_prefix_dir=`$ECHO "$destdir" | $SED -e "s%$destlibdir\$%%"`
 
 	  # Don't allow the user to place us outside of our expected
 	  # location b/c this prevents finding dependent libraries that
 	  # are installed to the same prefix.
 	  # At present, this check doesn't affect windows .dll's that
 	  # are installed into $libdir/../bin (currently, that works fine)
Index: libtool-2.4/libltdl/config/ltmain.m4sh
===================================================================
--- libtool-2.4.orig/libltdl/config/ltmain.m4sh	2010-09-22 18:45:18.000000000 +-0300
+++ libtool-2.4/libltdl/config/ltmain.m4sh	2010-09-22 18:45:18.000000000 +-0300
@@ -2161,14 +2161,18 @@
 
 	func_dirname "$file" "/" ""
 	dir="$func_dirname_result"
 	func_append dir "$objdir"
 
 	if test -n "$relink_command"; then
+          # Strip any trailing slash from the destination.
+          func_stripname '' '/' "$libdir"
+          destlibdir=$func_stripname_result
+
 	  # Determine the prefix the user has applied to our future dir.
-	  inst_prefix_dir=`$ECHO "$destdir" | $SED -e "s%$libdir\$%%"`
+	  inst_prefix_dir=`$ECHO "$destdir" | $SED -e "s%$destlibdir\$%%"`
 
 	  # Don't allow the user to place us outside of our expected
 	  # location b/c this prevents finding dependent libraries that
 	  # are installed to the same prefix.
 	  # At present, this check doesn't affect windows .dll's that
 	  # are installed into $libdir/../bin (currently, that works fine)
